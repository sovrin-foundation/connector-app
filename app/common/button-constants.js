// @flow
export const DENY = 'Deny'
export const CONNECT = 'Connect'
