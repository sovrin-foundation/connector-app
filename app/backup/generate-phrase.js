// @flow

import React, { PureComponent } from 'react'
import { Image, Dimensions } from 'react-native'
import { createStackNavigator } from 'react-navigation'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import type { Store } from '../store/type-store'
import type { ReactNavigation } from '../common/type-common'
import {
  Container,
  CustomView,
  CustomText,
  Icon,
  CustomButton,
  CustomHeader,
  Loader,
} from '../components'
import { isBiggerThanShortDevice } from '../common/styles'
import { genRecoveryPhraseRoute, verifyRecoveryPhraseRoute } from '../common'
import { color } from '../common/styles/constant'
import type {
  GenerateRecoveryPhraseProps,
  GenerateRecoveryPhraseState,
  Passphrase,
  ChatBubbleDimensions,
  PassphraseTextProps,
  PassphraseErrorProps,
} from './type-backup'
import { BACKUP_STORE_STATUS } from './type-backup'
import { generateRecoveryPhrase } from './backup-store'
import {
  RECOVERY_PHRASE_CLOSE_TEST_ID,
  SUBMIT_RECOVERY_PHRASE_TEST_ID,
  SUBMIT_RECOVERY_PHRASE_BUTTON_TITLE,
} from './backup-constants'
import styles, { chatBubbleTextOffset } from './styles'
import { getBackupPassphrase, getBackupStatus } from '../store/store-selector'
import { PASSPHRASE_GENERATION_ERROR } from '../common'

const closeImage = require('../images/iconClose.png')
const transparentBands = require('../images/transparentBands.png')
const textBubble = require('../images/textBubble.png')

const PassphraseLoader = () => {
  return (
    <CustomView style={[styles.genRecoveryPhraseLoadingContainer]}>
      <Loader delay={10} showMessage={false} size={30} />
    </CustomView>
  )
}

const PassphraseError = (props: PassphraseErrorProps) => {
  return (
    <CustomView
      style={[
        styles.genRecoveryPhraseContainer,
        {
          width:
            props.chatBubbleDimensions && props.chatBubbleDimensions.width
              ? props.chatBubbleDimensions.width - chatBubbleTextOffset
              : null,
        },
      ]}
    >
      <CustomText transparentBg darkgray center>
        {PASSPHRASE_GENERATION_ERROR}
      </CustomText>
    </CustomView>
  )
}

const PassphraseText = (props: PassphraseTextProps) => {
  return (
    <CustomView
      style={[
        styles.genRecoveryPhraseContainer,
        {
          //width of textContainer is chatBubbleDimensions width minus chatBubbleTextOffset
          //so that passPhrase text is not cutting beyond the chatBubble
          width:
            props.chatBubbleDimensions && props.chatBubbleDimensions.width
              ? props.chatBubbleDimensions.width - chatBubbleTextOffset
              : null,
        },
      ]}
    >
      <CustomText transparentBg style={[styles.genRecoveryPhrase]}>
        {props.recoveryPassphrase.phrase}
      </CustomText>
    </CustomView>
  )
}

export class GenerateRecoveryPhrase extends PureComponent<
  GenerateRecoveryPhraseProps,
  GenerateRecoveryPhraseState
> {
  state = {
    chatBubbleDimensions: {
      width: 0,
      height: 0,
      x: 0,
      y: 0,
    },
  }

  componentDidMount() {
    this.props.generateRecoveryPhrase()
  }

  verifyRecoveryPhrase = () => {
    const { navigation: { navigate, state } } = this.props

    navigate(verifyRecoveryPhraseRoute, {
      recoveryPassphrase: this.props.recoveryPassphrase.hash,
      initialRoute: state.params.initialRoute,
    })
  }

  static navigationOptions = ({ navigation }) => ({
    header: (
      <CustomHeader
        flatHeader
        largeHeader
        backgroundColor={color.bg.eleventh.color}
      >
        <CustomView />

        <CustomView style={[styles.headerSpacer]}>
          <Icon
            medium
            onPress={() => navigation.goBack(null)}
            testID={RECOVERY_PHRASE_CLOSE_TEST_ID}
            iconStyle={[styles.headerIcon]}
            src={closeImage}
          />
        </CustomView>
      </CustomHeader>
    ),
    gesturesEnabled: false,
  })

  //TODO fix refactor UI
  //first the image should be used as a ImageBackground component
  ImageContents = (
    recoveryStatus: string,
    recoveryPassphrase: Passphrase,
    chatBubbleDimensions: ChatBubbleDimensions
  ) => {
    if (recoveryStatus === BACKUP_STORE_STATUS.GENERATE_PHRASE_LOADING) {
      return <PassphraseLoader />
    }

    if (
      recoveryStatus === BACKUP_STORE_STATUS.GENERATE_PHRASE_FAILURE ||
      recoveryStatus === BACKUP_STORE_STATUS.GENERATE_BACKUP_FILE_FAILURE
    ) {
      // This block is where we need to try handling passphrase generation differently
      return <PassphraseError chatBubbleDimensions={chatBubbleDimensions} />
    }
    return (
      <PassphraseText
        recoveryPassphrase={recoveryPassphrase}
        chatBubbleDimensions={chatBubbleDimensions}
      />
    )
  }

  setChatBubbleDimensions = (chatBubbleDimensions: ChatBubbleDimensions) => {
    this.setState({
      chatBubbleDimensions,
    })
  }

  render() {
    const { recoveryPassphrase, recoveryStatus } = this.props
    const disableButton =
      recoveryStatus === BACKUP_STORE_STATUS.GENERATE_PHRASE_FAILURE ||
      recoveryStatus === BACKUP_STORE_STATUS.GENERATE_PHRASE_LOADING ||
      recoveryStatus === BACKUP_STORE_STATUS.GENERATE_BACKUP_FILE_FAILURE

    return (
      <Container style={[styles.genRecovery]} safeArea>
        <Image source={transparentBands} style={[styles.backgroundImage]} />
        <Container>
          <CustomView>
            <CustomText
              transparentBg
              numberOfLines={1}
              style={[styles.genRecoveryText]}
            >
              Recovery Phrase generated
            </CustomText>
          </CustomView>
          <CustomView center>
            <CustomText
              center
              transparentBg
              style={[styles.genRecoveryMessage]}
            >
              This is your Recovery Phrase. Write it down, and don't share it
              with anyone.
            </CustomText>
          </CustomView>
          <CustomView center>
            <Image
              source={textBubble}
              style={[styles.imageIcon]}
              onLayout={event =>
                this.setChatBubbleDimensions(event.nativeEvent.layout)
              }
            />
            {this.ImageContents(
              recoveryStatus,
              recoveryPassphrase,
              this.state.chatBubbleDimensions
            )}
          </CustomView>
          <CustomView center>
            <CustomText
              center
              transparentBg
              style={[styles.genRecoverySecondMessage]}
            >
              You will need this Recovery Phrase to restore from a backup. Keep
              it safe.
            </CustomText>
          </CustomView>
        </Container>
        <CustomView>
          <CustomView center>
            <CustomText
              transparentBg
              center
              style={[styles.genRecoverySmallMessage]}
            >
              Are you sure you wrote it down?
            </CustomText>
          </CustomView>
          <CustomButton
            disabled={disableButton}
            large={isBiggerThanShortDevice ? true : false}
            onPress={this.verifyRecoveryPhrase}
            testID={SUBMIT_RECOVERY_PHRASE_TEST_ID}
            style={[styles.submitButton]}
            customColor={{
              color: color.bg.eleventh.color,
              fontWeight: '600',
              fontSize: 18,
            }}
            title={SUBMIT_RECOVERY_PHRASE_BUTTON_TITLE}
          />
        </CustomView>
      </Container>
    )
  }
}

const mapStateToProps = (state: Store) => {
  return {
    recoveryPassphrase: getBackupPassphrase(state),
    recoveryStatus: getBackupStatus(state),
  }
}
const mapDispatchToProps = dispatch =>
  bindActionCreators({ generateRecoveryPhrase }, dispatch)

export default createStackNavigator({
  [genRecoveryPhraseRoute]: {
    screen: connect(mapStateToProps, mapDispatchToProps)(
      GenerateRecoveryPhrase
    ),
  },
})
