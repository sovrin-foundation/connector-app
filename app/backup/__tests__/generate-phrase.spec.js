// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import { GenerateRecoveryPhrase } from '../generate-phrase'
import { getNavigation } from '../../../__mocks__/static-data'
import { settingsRoute } from '../../common'
import { BACKUP_STORE_STATUS } from '../type-backup'

describe('<GenerateRecoveryPhrase />', () => {
  const recoveryPassphrase = 'hello some passphrase'
  const navigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
    state: {
      params: {
        recoveryPassphrase,
        initialRoute: settingsRoute,
      },
    },
  }

  it('should match snapshot', () => {
    const tree = renderer.create(
      <GenerateRecoveryPhrase
        recoveryPassphrase={{
          phrase: recoveryPassphrase,
          salt: 'salt',
          hash: 'hash',
        }}
        generateRecoveryPhrase={jest.fn()}
        recoveryStatus={BACKUP_STORE_STATUS.GENERATE_PHRASE_SUCCESS}
        navigation={navigation}
      />
    )
    tree.getInstance().setState({
      chatBubbleDimensions: {
        width: 360,
        height: 199,
        x: 0,
        y: 0,
      },
    })
    expect(tree.toJSON()).toMatchSnapshot()
  })
  it('should match snapshot for loading state', () => {
    const tree = renderer.create(
      <GenerateRecoveryPhrase
        recoveryPassphrase={{
          phrase: recoveryPassphrase,
          salt: 'salt',
          hash: 'hash',
        }}
        generateRecoveryPhrase={jest.fn()}
        recoveryStatus={BACKUP_STORE_STATUS.GENERATE_PHRASE_LOADING}
        navigation={navigation}
      />
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })
  it('should match snapshot for error state', () => {
    const tree = renderer.create(
      <GenerateRecoveryPhrase
        recoveryPassphrase={{
          phrase: recoveryPassphrase,
          salt: 'salt',
          hash: 'hash',
        }}
        generateRecoveryPhrase={jest.fn()}
        recoveryStatus={BACKUP_STORE_STATUS.GENERATE_PHRASE_FAILURE}
        navigation={navigation}
      />
    )
    tree.getInstance().setState({
      chatBubbleDimensions: {
        width: 360,
        height: 199,
        x: 0,
        y: 0,
      },
    })

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
