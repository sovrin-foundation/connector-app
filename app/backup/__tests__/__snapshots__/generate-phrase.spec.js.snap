// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenerateRecoveryPhrase /> should match snapshot 1`] = `
<RCTSafeAreaView
  safeArea={true}
  style={
    Object {
      "backgroundColor": "#4A90E2",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "fontWeight": "600",
              "marginBottom": "2%",
            },
          ]
        }
      >
        Recovery Phrase generated
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 5,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        This is your Recovery Phrase. Write it down, and don't share it with anyone.
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        onLayout={[Function]}
        source={1}
        style={
          Array [
            Object {
              "marginBottom": 0,
              "resizeMode": "stretch",
              "width": "100%",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "left": 20,
            "paddingHorizontal": "7%",
            "position": "absolute",
            "top": 20,
            "width": 320,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "color": "#4A90E2",
                "fontSize": 18,
                "fontWeight": "900",
                "lineHeight": 28,
                "textAlign": "center",
              },
            ]
          }
        >
          hello some passphrase
        </Text>
      </View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 20,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        You will need this Recovery Phrase to restore from a backup. Keep it safe.
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "lineHeight": 17,
              "marginBottom": "3%",
            },
          ]
        }
      >
        Are you sure you wrote it down?
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="submit-recovery-passphrase"
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "borderRadius": 5,
                "elevation": 7,
                "height": 43,
                "marginBottom": 15,
                "marginLeft": "2%",
                "marginRight": "2%",
                "shadowColor": "#777777",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 5,
              },
            ],
          ]
        }
        testID="submit-recovery-passphrase"
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
                Object {
                  "borderRadius": 5,
                  "elevation": 7,
                  "height": 43,
                  "marginBottom": 15,
                  "marginLeft": "2%",
                  "marginRight": "2%",
                  "shadowColor": "#777777",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                },
              ],
              false,
              false,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A90E2",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  Object {
                    "fontSize": 18,
                  },
                  undefined,
                  Object {
                    "fontWeight": "600",
                  },
                  undefined,
                  false,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<GenerateRecoveryPhrase /> should match snapshot for error state 1`] = `
<RCTSafeAreaView
  safeArea={true}
  style={
    Object {
      "backgroundColor": "#4A90E2",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "fontWeight": "600",
              "marginBottom": "2%",
            },
          ]
        }
      >
        Recovery Phrase generated
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 5,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        This is your Recovery Phrase. Write it down, and don't share it with anyone.
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        onLayout={[Function]}
        source={1}
        style={
          Array [
            Object {
              "marginBottom": 0,
              "resizeMode": "stretch",
              "width": "100%",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "left": 20,
            "paddingHorizontal": "7%",
            "position": "absolute",
            "top": 20,
            "width": 320,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "color": "#4A4A4A",
              },
            ]
          }
        >
          Uh oh... Looks like something went wrong.
        </Text>
      </View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 20,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        You will need this Recovery Phrase to restore from a backup. Keep it safe.
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "lineHeight": 17,
              "marginBottom": "3%",
            },
          ]
        }
      >
        Are you sure you wrote it down?
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="submit-recovery-passphrase"
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "borderRadius": 5,
                "elevation": 7,
                "height": 43,
                "marginBottom": 15,
                "marginLeft": "2%",
                "marginRight": "2%",
                "shadowColor": "#777777",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 5,
              },
            ],
          ]
        }
        testID="submit-recovery-passphrase"
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
                Object {
                  "borderRadius": 5,
                  "elevation": 7,
                  "height": 43,
                  "marginBottom": 15,
                  "marginLeft": "2%",
                  "marginRight": "2%",
                  "shadowColor": "#777777",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                },
              ],
              Object {
                "backgroundColor": "#dadee0",
              },
              Array [
                Object {
                  "backgroundColor": "#ffffff",
                },
                Object {
                  "opacity": 0.4,
                },
              ],
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A90E2",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  Object {
                    "fontSize": 18,
                  },
                  undefined,
                  Object {
                    "fontWeight": "600",
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<GenerateRecoveryPhrase /> should match snapshot for loading state 1`] = `
<RCTSafeAreaView
  safeArea={true}
  style={
    Object {
      "backgroundColor": "#4A90E2",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "fontWeight": "600",
              "marginBottom": "2%",
            },
          ]
        }
      >
        Recovery Phrase generated
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 5,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        This is your Recovery Phrase. Write it down, and don't share it with anyone.
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        onLayout={[Function]}
        source={1}
        style={
          Array [
            Object {
              "marginBottom": 0,
              "resizeMode": "stretch",
              "width": "100%",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "padding": 40,
            "position": "absolute",
            "top": 0,
          }
        }
      />
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 20,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        You will need this Recovery Phrase to restore from a backup. Keep it safe.
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "lineHeight": 17,
              "marginBottom": "3%",
            },
          ]
        }
      >
        Are you sure you wrote it down?
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="submit-recovery-passphrase"
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "borderRadius": 5,
                "elevation": 7,
                "height": 43,
                "marginBottom": 15,
                "marginLeft": "2%",
                "marginRight": "2%",
                "shadowColor": "#777777",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 5,
              },
            ],
          ]
        }
        testID="submit-recovery-passphrase"
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
                Object {
                  "borderRadius": 5,
                  "elevation": 7,
                  "height": 43,
                  "marginBottom": 15,
                  "marginLeft": "2%",
                  "marginRight": "2%",
                  "shadowColor": "#777777",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                },
              ],
              Object {
                "backgroundColor": "#dadee0",
              },
              Array [
                Object {
                  "backgroundColor": "#ffffff",
                },
                Object {
                  "opacity": 0.4,
                },
              ],
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A90E2",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  Object {
                    "fontSize": 18,
                  },
                  undefined,
                  Object {
                    "fontWeight": "600",
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
